.github/workflows/dotnet.yml

```diff
@@ -9,6 +9,23 @@ on:
       - master
 
 jobs:
+  secret-scan:
+    runs-on: ubuntu-latest
+    
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Run Gitleaks
+        uses: gitleaks/gitleaks-action@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I'm not comfortable with passing GITHUB_TOKEN. Is there any safer method for secret scanning?